assets/src/apps/authoring/components/RightMenu/RightMenu.tsx

```diff
@@ -3,38 +3,66 @@ import { Tab, Tabs } from 'react-bootstrap';
 import { useSelector } from 'react-redux';
 import { selectCurrentActivity } from '../../../delivery/store/features/activities/slice';
 import { selectState as selectPageState } from '../../store/page/slice';
-import PropertyEditor, { Schema } from '../PropertyEditor/PropertyEditor';
-import lessonSchema from '../PropertyEditor/schemas/lesson';
-import screenSchema from '../PropertyEditor/schemas/screen';
+import PropertyEditor from '../PropertyEditor/PropertyEditor';
+import lessonSchema, {getLessonData, lessonUiSchema} from '../PropertyEditor/schemas/lesson';
+import screenSchema, {getScreenData, screenUiSchema} from '../PropertyEditor/schemas/screen';
+import { JSONSchema7 } from 'json-schema';
 
 const RightMenu: React.FC<any> = (props) => {
   const [selectedTab, setSelectedTab] = useState<string>('lesson');
   const currentActivity = useSelector(selectCurrentActivity);
   const currentLesson = useSelector(selectPageState);
 
   // TODO: dynamically load schema from Part Component configuration
-  const componentSchema: Schema = { type: 'object' };
+  const componentSchema: JSONSchema7 = { type: 'object' };
   const currentComponent = null;
-
+  const screenData = getScreenData(currentActivity?.model.custom);
+  const lessonData = getLessonData(props.content);
```

Comment:
oh, why do we need to send in this "content" into props? there is the `currentLesson` on line 14 if it doesn't have what we need in there, then I think we can add it at the redux level so it will

Reply:
ohkk then...